### PR TITLE
feat: Add support for configuring wasmCloud Secrets

### PR DESCRIPTION
**Pull request by joonas**
## Feature or Problem

Since the features support is rolling out in the wasmCloud host, we should also add support to use it in the operator.

### CODE REVIEW

**Review by ricochet — APPROVED** [review]

File: examples/full-config/wasmcloud-annotated.yaml [hunk 1]
```diff
@@ -68,6 +68,7 @@ spec:
     traces:
       enable: false
       endpoint: "traces-specific-otel-collector.svc"
+  secrets_topic_prefix: "wasmcloud.secrets"
```
Comment (lxfontes) [inline examples/full-config/wasmcloud-annotated.yaml:71]: add `Optional/Required:` + decription to the key

**Review by lxfontes — CHANGES_REQUESTED** [review]
small doc nit

**Review by protochron — APPROVED** [review]

**Review by lxfontes — APPROVED** [review]

File: examples/full-config/wasmcloud-annotated.yaml [hunk 2]
```diff
@@ -68,6 +68,9 @@ spec:
     traces:
       enable: false
       endpoint: "traces-specific-otel-collector.svc"
+  # Optional: Subject prefix that will be used by the host to query for wasmCloud Secrets.
+  # See https://wasmcloud.com/docs/concepts/secrets for more context
+  secrets_topic_prefix: "wasmcloud.secrets"
```
Comment (lxfontes) [inline examples/full-config/wasmcloud-annotated.yaml:73]: heads up this will actually be `secretsTopicPrefix` due to camel-casing in the struct.

**Comment by joonas** [comment]
Holding off merging this until we can change the `wadm-*` pins to point to non `-beta` versions of the published crates.

**Pull request merged**